scripts/e2e_verify/main.py

```
@@ -0,0 +1,252 @@
+
+# Timeout configuration in milliseconds
+DOWNLOAD_TIMEOUT = 5 * 60 * 1000  # 5 minutes
+HASH_TIMEOUT = 60 * 1000  # 1 minute
+PRECHECK_START_TIMEOUT = 10 * 60 * 1000  # 10 minutes
+PRECHECK_COMPLETE_TIMEOUT = 4 * 60 * 60 * 1000  # 4 hours
+
+
+def log(step: int, total: int, message: str) -> None:
+    """Print timestamped log message."""
+    timestamp = datetime.now(UTC).strftime("%H:%M:%S")
+    print(f"[{timestamp}] Step {step}/{total}: {message}")  # noqa: T201
```

Ruff (0.14.7)

Unused `noqa` directive (non-enabled: `T201`)

Remove unused `noqa` directive

(RUF100)